src/main/java/org/generationcp/middleware/service/DataImportServiceImpl.java

```diff
@@ -434,13 +435,19 @@ public void processExperimentalDesign(final Workbook workbook, final String prog
 	public void addEntryTypeVariableIfNotExists(final Workbook workbook,
 		final String programUUID) {
 
-		final boolean entryTypeIdExists = this.findMeasurementVariableByTermId(TermId.ENTRY_TYPE.getId(), workbook.getFactors()).isPresent();
+		final Optional<MeasurementVariable> optEntryType = this.findMeasurementVariableByTermId(TermId.ENTRY_TYPE.getId(), workbook.getFactors());
+		final boolean entryTypeidExistsInRow = this.findMeasurementRowByTermId(TermId.ENTRY_TYPE.getId(), workbook.getObservations()).isPresent();
 
-		// If ENTRY_TYPE variable is not existing in both Condition and Factors Section of workbook
-		// Automatically add ENTRY_TYPE variable as it is required in creating a new Study.
-		if (!entryTypeIdExists) {
-			final MeasurementVariable entryType = this.createMeasurementVariable(TermId.ENTRY_TYPE.getId(), null, Operation.ADD, PhenotypicType.GERMPLASM, programUUID);
-			workbook.getFactors().add(entryType);
+		if (!optEntryType.isPresent() || !entryTypeidExistsInRow) {
+			final MeasurementVariable entryType;
+			if (!optEntryType.isPresent()) {
+				// If ENTRY_TYPE variable is not existing in Factors Section of workbook
+				// Automatically add ENTRY_TYPE variable as it is required in creating a new Study.
+				entryType = this.createMeasurementVariable(TermId.ENTRY_TYPE.getId(), null, Operation.ADD, PhenotypicType.GERMPLASM, programUUID);
+				workbook.getFactors().add(entryType);
+			} else {
+				entryType = optEntryType.get();
+			}
```

Comment:
What if there are only some, not all, rows that don't have ENTRY_TYPE value? All of them will be overwritten to have TEST entry type, right? (if so, that is not correct behavior)

Reply:
Better to find only the rows without ENTRY_TYPE data and update those to have TEST_ENTRY value